composer/utils/parallelism.py

```diff
@@ -41,6 +40,24 @@ class FSDPConfig:
     use_orig_params: bool = True
     verbose: bool = False
 
+    _device_mesh: Optional[DeviceMesh] = field(default=None, init=False, repr=False)
+
+    def __init__(self, **kwargs):
+        if 'device_mesh' in kwargs or '_device_mesh' in kwargs:
+            raise ValueError(
+                f'Directly specifying device mesh for FSDP was deprecated in Composer version 0.24.0. Please specify \'data_parallel_shard_degree\' and/or \'data_parallel_replicate_degree\' instead.',
+            )
```

Comment:
```suggestion
                f"Directly specifying device mesh for FSDP was deprecated in Composer version 0.24.0. Please specify 'data_parallel_shard_degree' and/or 'data_parallel_replicate_degree' instead.",
            )
```
nit: mildly easier

Reply:
long line with this error msg was bothering me so resolved this as part of that